pkg/api/types.go

```diff
@@ -758,12 +758,12 @@ type NodeSpec struct {
 
 // NodeStatus is information about the current status of a node.
 type NodeStatus struct {
-	// Queried from cloud provider, if available.
-	HostIP string `json:"hostIP,omitempty"`
 	// NodePhase is the current lifecycle phase of the node.
 	Phase NodePhase `json:"phase,omitempty"`
 	// Conditions is an array of current node conditions.
 	Conditions []NodeCondition `json:"conditions,omitempty"`
+	// Queried from cloud provider, if available.
+	Addresses []NodeAddress `json:"addresses,omitempty"`
```

Comment:
How about AlternateEndpoints -- so, including port?

We've discussed allowing hosts in Endpoints, in #4370.

I want Nodes to actually populate an Endpoints object, as discussed in #4440.

Endpoints will need to be able to carry various information (e.g., internal vs. external), as discussed in #4482 and elsewhere.